pkg/util/version/version.go

```diff
@@ -58,3 +68,26 @@ func (v *Version) Lt(w *Version) bool {
 
 	return false
 }
+
+func (v *Version) Eq(w *Version) bool {
+	return bytes.Compare(v.V[:], w.V[:]) == 0
+}
+
+func GetUpgrade(v *Version) (*Upgrade, error) {
+	for _, u := range Upgrades {
+		if u.Version.V[0] == v.V[0] &&
```

Comment:
Because we should have only 1 upgrade path perversion (as we have now).
Let's say we have `4.4.6` and `4.3.20` configured on RP side as our upgrade paths.
So if the cluster is incoming with `4.4.4` - we will return `4.4.6` as an upgrade path. And we have code in upgrade.go to prevent downgrades if customer is already on higher version than our configures upgrade path.
```
if !desired.Lt(upgrade.Version) {
			i.log.Printf("not upgrading: cvo desired version is %s", cv.Status.Desired.Version)
			return nil
		}
```
This is same behaviour as we had before just with multiple versions. The challenge here is that we don't know How we want to handle multiple versions and this just keeps existing behaviour :/

Reply:
Thanks for explanation the first point. I think it is just name that is confusing in this case: in English I read it as "give me an upgrade object for version `v`", but in fact I can get and upgrade with version != `v`.

Mabe rename it to "get next patch upgrade" or something like that?

I tihnk you covered a bit the second point [here](https://github.com/Azure/ARO-RP/pull/664#discussion_r437316086):

> We could have one more test to check if we have only single x.Y version path configured. But I think this all code base should go away soon when we decide what we want to do with our version management,

I think it worth adding a test to check that there are no two 4.4.* or 4.3.* versions at the same time. After migrating from 4.3 to 4.4 we will have 4.4 -> 4.5 at some point. Right? I suspect that we will want to keep the code that manages versions.

Reply:
Added the test. But I'm very specific about future of this code past 4.4 :D